addons/sourcemod/scripting/AdminLogging.sp

```diff
@@ -50,18 +50,71 @@ public Action OnLogAction(Handle source, Identity ident, int client, int target,
 
 	char szWebhookURL[1000];
 	g_cvWebhook.GetString(szWebhookURL, sizeof szWebhookURL);
-
+	if(!szWebhookURL[0])
+	{
+		LogError("[Adminlogging] No webhook found or specified.");
+		return Plugin_Handled;
+	}
+	
 	Webhook webhook = new Webhook(sMessage);
-	webhook.Execute(szWebhookURL, OnWebHookExecuted);
-	delete webhook;
+	
+	DataPack pack = new DataPack();
+	pack.WriteCell(view_as<int>(webhook));
+	pack.WriteString(szWebhookURL);
+	
+	webhook.Execute(szWebhookURL, OnWebHookExecuted, pack);
 
 	return Plugin_Handled;
 }
 
 public void OnWebHookExecuted(HTTPResponse response, DataPack pack)
 {
-    if (response.Status != HTTPStatus_OK)
-    {
-        LogError("Failed to send adminlogging webhook");
-    }
+	static int retries;
+	
+	pack.Reset();
+	Webhook hook = view_as<Webhook>(pack.ReadCell());
+
+	if (response.Status != HTTPStatus_OK)
+	{
+		if(retries < 3)
+			PrintToServer("[AdminLogging] Failed to send the webhook. Resending it .. (%d/3)", retries);
+		
+		else if(retries >= 3)
+		{
+			LogError("[AdminLogging] Could not send the webhook after %d retries.", retries);
+			delete hook;
+			delete pack;
+			return;
+		}
+		
+		char webhookURL[PLATFORM_MAX_PATH];
+		pack.ReadString(webhookURL, sizeof(webhookURL));
+		
+		DataPack newPack;
+		newPack.WriteCell(view_as<int>(hook));
+		newPack.WriteString(webhookURL);
+		CreateDataTimer(0.5, ExecuteWebhook_Timer, newPack);
```

Comment:
mb you can put back this function before calling writecell

i didnt know that CreateDataTimer was actually creating the datapack:
- https://github.com/alliedmodders/sourcemod/blob/6e73aba2500f1565464955f48f17a4ed52da2a5c/plugins/include/timers.inc#L195